### DISCUSSION TITLE
What is the endpoint url and username for Next Auth with credentials?

### DISCUSSION THREAD

**Discussion by SamnangLam** [hasura/graphql-engine#7666] [Question]
I've blocked with Next auth by credentials.

On Hasura console, I'm seeing Graphql endpoint is `https://***.hasura.app/v1/graphql`
So I deleted `v1/graphql` and added it. Please check the screenshot.
![image](https://user-images.githubusercontent.com/45047729/136533539-34a51ba0-4960-4534-98d9-2b64afd85478.png)

Then I'm getting 404 error. What is the correct endpoint for credential?
Also, I'm not sure what is the username on Hasura console and where can I find it.
![image](https://user-images.githubusercontent.com/45047729/136533328-bdf16b8e-9191-4abe-931d-4694b1c9edb9.png)

Please help me!!!
Thanks.

**Comment by BenoitRanque** [comment]
Hi!

I'm not sure how you expect hasura to behave in this case.
Could you point me to the docs/blog/article that lead you to think this is possible?

Hasura does not handle the login part of authentication.
Hasura can be configured to validate a JWT, or send a request to a webhook and use that to validate request.

Both of these are intended to be used once a session has already been aquired, how that session is aquired is left to the user to implement.

In your code above, it seems you'd expect hasura to know how to create a session, given a username and password.
Hasura does not know how to handle this. What table should hasura look in? What columns should be looked at?
What claims should be issued? What roles should be associated with the user
These are all questions hasura does not have answers for.

What you should probably look into is building a custom action, or http endpoint, that will know how to create a session, given the right input.

**Reply by SamnangLam** [reply]
Thanks for your reply.
